src/jit/regalloc.cpp

```diff
@@ -698,8 +700,15 @@ regNumber     Compiler::raUpdateRegStateForArg(RegState *regState, LclVarDsc *ar
         }
         else
         {
+#ifndef ARM_SOFTFP
             assert(regState->rsIsFloat);
             assert(emitter::isDoubleReg(inArgReg));
+#else
+            // The same as for long for softfp abi
+            assert((inArgReg == REG_R0) || (inArgReg == REG_R2));
+            assert(!regState->rsIsFloat);
+            regState->rsCalleeRegArgMaskLiveIn |= genRegMask((regNumber)(inArgReg+1));
+#endif
         }
```

Comment:
this change isn't necessary. above, change "if (info.compIsVarArgs)" to "if (info.compIsVarArgs || opts.compUseSoftFP)"